tuf/client/updater.py

```diff
@@ -2721,20 +2842,15 @@ def download_target(self, target, destination_directory):
                                target_filepath.lstrip(os.sep))
     destination = os.path.abspath(destination)
     target_dirpath = os.path.dirname(destination)
-    if target_dirpath:
-      try:
-        os.makedirs(target_dirpath)
-      
-      except OSError as e:
-        if e.errno == errno.EEXIST:
-          pass
-        
-        else:
-          raise
     
-    else:
-      message = repr(target_dirpath) + ' does not exist.'
-      logger.warning(message)
-      raise tuf.Error(message)
+    try:
+      os.makedirs(target_dirpath)
+    
+    except OSError as e:
+      if e.errno == errno.EEXIST:
```

Comment:
Note: `errno.EEXIST` is only raised when the full dir to be `makedirs`ed already exists (e.g. `makedirs("foo/bar/baz")`, where "foo/bar/baz" was there already).

If only parts of the required directory structure exist, this is not an error, and the deeper-nested parts are created within the outer existing structure (e.g. "foo/bar/baz/frobble" inside of "foo/bar").